src/envoy/utils/mixer_control.h

```diff
@@ -42,5 +48,40 @@ Grpc::AsyncClientFactoryPtr GrpcClientFactoryForCluster(
     const std::string &cluster_name, Upstream::ClusterManager &cm,
     Stats::Scope &scope);
 
+std::unique_ptr<const LocalAttributes> CreateLocalAttributes(
+    const LocalNode &local);
+
+bool Extract(const envoy::api::v2::core::Node &node, LocalNode *args);
+
+inline bool ReadMap(
```

Comment:
I don't think this is a correct use of metadata in envoy. It assumes that the key is some filter ID, and the list of values is inside the struct value. Take a look at `source/common/config/metadata.cc`.

Reply:
node metadata is actually `google.protobuf.struct` and it does not assume anything about a filter. This is too high level.

Reply:
are you suggesting that this should be nested inside Mixer ?

Reply:
OK, yes, I assumed node proto is filter-aware. This is fine.